src/cargo/util/rustc.rs

```diff
@@ -28,6 +28,8 @@ pub struct Rustc {
     pub version: semver::Version,
     /// The host triple (arch-platform-OS), this comes from verbose_version.
     pub host: InternedString,
+    /// The rustc full commit hash, this comes from `verbose_version`.
+    pub commit_hash: Option<String>,
```

Comment:
Could you update the commit message for this to say why this is added?  It took me digging 4 commits in further and searching to find it.